lib/media/segment_prefetch.js

```diff
@@ -0,0 +1,173 @@
+/*! @license
+ * Shaka Player
+ * Copyright 2016 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+goog.require('shaka.net.NetworkingEngine');
+goog.require('shaka.media.InitSegmentReference');
+goog.require('shaka.media.SegmentReference');
+goog.provide('shaka.media.SegmentPrefetch');
+goog.require('shaka.log');
+
+/**
+ * @summary
+ * This class manages segment prefetch operations.
+ * Called by StreamingEngine to prefetch next N segments
+ * ahead of playhead, to reduce the chances of rebuffering.
+ */
+shaka.media.SegmentPrefetch = class {
+  /**
+   * @param {number} prefetchLimit
+   * @param {shaka.extern.Stream} stream
+   * @param {function(
+   *  !(shaka.media.InitSegmentReference|shaka.media.SegmentReference),
+   *  shaka.extern.Stream
+   * ):!shaka.net.NetworkingEngine.PendingRequest} fetchDispatcher
+   */
+  constructor(prefetchLimit, stream, fetchDispatcher) {
+    /**
+     * @private {number}
+     */
+    this.prefetchLimit_ = prefetchLimit;
+
+    /** @private {shaka.extern.Stream} */
+    this.stream_ = stream;
+
+    /** @private {number} */
+    this.prefetchPosTime_ = 0;
+
+    /** @private {function(
+     * !(shaka.media.InitSegmentReference|shaka.media.SegmentReference),
+     * shaka.extern.Stream
+     * ):!shaka.net.NetworkingEngine.PendingRequest} */
+    this.fetchDispatcher_ = fetchDispatcher;
+
+    /**
+     * @private {!Map.<shaka.media.SegmentReference,
+                 !shaka.net.NetworkingEngine.PendingRequest>}
+     */
+    this.segmentPrefetchMap_ = new Map();
+  }
+
+  /**
+   * Fetch next segments ahead of current segment.
+   *
+   * @param {(!shaka.media.SegmentReference)} startReference
+   * @public
+   */
+  prefetchSegments(startReference) {
+    if (!(this.prefetchLimit_ > 0)) {
```

Comment:
`prefetchLimit_` is non-nullable, so couldn't this just be:
```js
if (this.prefetchLimit_ <= 0) {
```

Reply:
replaced with assert